client/state/themes/reducer.js

```diff
@@ -227,6 +226,10 @@ export const queries = ( () => {
 		[ THEMES_RECEIVE ]: ( state, { siteId, themes } ) => {
 			return applyToManager( state, siteId, 'receive', true, themes );
 		},
+		[ THEME_REQUEST_FAILURE ]: ( state, { siteId, themeId, error } ) => {
+			const theme = [ { id: themeId, error } ];
+			return applyToManager( state, siteId, 'receive', true, theme );
+		},
```

Comment:
I wonder if the bug I found is related to this, actually. If we need to guard the theme sheet against the case where no theme object has been fetched, can't we just check if the theme object prop is falsey and `renderError` if that is the case?